scripts/plan_cost_runner.py

```diff
@@ -50,14 +50,40 @@ def init_db(cli, dbname, benchmark_dir):
     print("INITIALIZATION DONE")
 
 
+class PlanCost:
+    def __init__(self):
+        self.total = 0
+        self.build_side = 0
+        self.probe_side = 0
+        self.time = 0
+
+    def __add__(self, other):
+        self.total += other.total
+        self.build_side += other.build_side
+        self.probe_side += other.probe_side
+        return self
+
+
 def op_inspect(op):
-    cost = 0
+    cost = PlanCost()
+    if op['name'] == "Query":
+        cost.time = op['timing']
     if op['name'] == 'HASH_JOIN' and not op['extra_info'].startswith('MARK'):
-        cost = op['cardinality']
-    if 'children' not in op:
+        cost.total = op['cardinality']
+        if 'cardinality' in op['children'][0]:
```

Comment:
You could use this
```py
children = {
	'probe_side': op['children'][0],
	'build_side': op['children'][1]
}

for side in children:
	child = children[side]
	if 'cardinality' in child:
		cost.__getattribute__(side) += child['cardinality']
	child_cost = op_inspect(child)
	cost += child_cost
```

Debatable whether this is more readable though

Reply:
That's essentially this:
```py
        left_cost = op_inspect(op['children'][0])
        right_cost = op_inspect(op['children'][1])
        cost.probe_side += left_cost.probe_side + right_cost.probe_side
        cost.build_side += left_cost.build_side + right_cost.build_side
        cost.total += left_cost.total + right_cost.total
```

With a little weirdness at the top because of the switch between probe_side/build_side